exonum/src/sandbox/consensus/round_details.rs

```diff
@@ -1631,6 +1714,37 @@ fn handle_tx_ignore_existing_tx_in_blockchain() {
     sandbox.add_time(Duration::from_millis(0));
 }
 
+/// Ignore transactions that fail `BlockchainMut::check_tx`.
+/// Idea of test is to receive incorrect tx (which is expected to be ignored) and
+/// then broadcast prevote without this tx.
+#[test]
+fn handle_tx_ignore_incorrect_tx() {
+    let sandbox = timestamping_sandbox();
+    let sandbox_state = SandboxState::new();
+
+    add_one_height(&sandbox, &sandbox_state);
+    sandbox.assert_state(Height(2), Round(1));
+
+    // add rounds & become leader
+    sandbox.add_time(Duration::from_millis(sandbox.current_round_timeout()));
+    assert!(sandbox.is_leader());
+
+    // Create correct tx, and then sign with the wrong destination.
+    let (pk, sk) = gen_keypair();
+    let incorrect_tx = gen_unverified_timestamping_tx().sign(TimestampingService::ID + 1, pk, &sk);
+
+    // Receive that message.
+    sandbox.recv(&incorrect_tx);
```

Comment:
Nit: Is it possible to check that the incorrect transaction is not in the mempool here?

Reply:
Well, it's not added to the pool at `recv` step, but to the tx cache. Added a check for that.